### ISSUE TITLE
signeR issue

### ISSUE THREAD

**Issue by TMante** [rvalieris/signeR#3]
Hi,

I'm encountering this very same issue with signeR as ulah (commented May 24th) eventhough I'm using the version 1.2.2 of signeR and R version 3.4.1. Any suggestions on how to overcome this?

> signatures <- signeR(M=mut, Opport=opp)
Error in (function (x)  : 
  unused arguments (model = list(model = "NMFstd", rank = 1, target = 0), method = "random")

Thank you very much,

**Comment by rvalieris** [comment]
Hello, 

which version of bioconductor are you using ? (output of library(BiocInstaller) )
also, include the output of sessionInfo() after loading the signeR package.

**Comment by ulah** [comment]
Hi rvalieris, Hi TMante,

there seems to be a conflict between required and masked libraries from signeR (SR) and VariantAnnotation (VA). As the error lists NMF, is suspect "nrun" and/or "seed", but I could be completely wrong ;)

So, here is what I tried to get signeR working for my data:

1) If you run the vignette w/o loading VA, the above mentioned error does not appear. However, you will not be able to run your own data as "readVcf" is a part of VA.
2) If you load SR first and then VA, the error appears. During dependency loading, you'll see a lot of packages being masked by VA - among others, 2 functions from NMF:

```
[...]

The following object is masked from ‘package:NMF’:

    nrun

[...]

The following object is masked from ‘package:NMF’:

    seed

[...]
```

However, if you load VA first and SR second, those functions are masked in the other direction during loading of SR:

```
Attaching package: ‘NMF’

The following object is masked from ‘package:DelayedArray’:

    seed

The following object is masked from ‘package:S4Vectors’:

    nrun

```

And now, everything works fine :)
Cheers,
Urs

**Comment by ulah** [comment]
btw, for completeness:

```
> sessionInfo()
R version 3.4.2 (2017-09-28)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.3 LTS

Matrix products: default
BLAS: /usr/lib/libblas/libblas.so.3.6.0
LAPACK: /usr/lib/lapack/liblapack.so.3.6.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=de_DE.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=de_DE.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=de_DE.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=de_DE.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] BSgenome.Hsapiens.UCSC.hg19_1.4.0 BSgenome_1.46.0                  
 [3] rtracklayer_1.38.0                signeR_1.4.0                     
 [5] NMF_0.20.6                        cluster_2.0.6                    
 [7] rngtools_1.2.4                    pkgmaker_0.22                    
 [9] registry_0.3                      VariantAnnotation_1.24.2         
[11] Rsamtools_1.30.0                  Biostrings_2.46.0                
[13] XVector_0.18.0                    SummarizedExperiment_1.8.0       
[15] DelayedArray_0.4.1                matrixStats_0.52.2               
[17] Biobase_2.38.0                    GenomicRanges_1.30.0             
[19] GenomeInfoDb_1.14.0               IRanges_2.12.0                   
[21] S4Vectors_0.16.0                  BiocGenerics_0.24.0              

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.14             lattice_0.20-35          class_7.3-14            
 [4] prettyunits_1.0.2        assertthat_0.2.0         digest_0.6.12           
 [7] foreach_1.4.3            gridBase_0.4-7           R6_2.2.2                
[10] plyr_1.8.4               RSQLite_2.0              ggplot2_2.2.1           
[13] zlibbioc_1.24.0          rlang_0.1.4              GenomicFeatures_1.30.0  
[16] progress_1.1.2           lazyeval_0.2.1           nloptr_1.0.4            
[19] blob_1.1.0               Matrix_1.2-12            RMySQL_0.10.13          
[22] BiocParallel_1.12.0      stringr_1.2.0            RCurl_1.95-4.8          
[25] bit_1.1-12               biomaRt_2.34.0           munsell_0.4.3           
[28] compiler_3.4.2           tibble_1.3.4             GenomeInfoDbData_0.99.1 
[31] codetools_0.2-15         XML_3.98-1.9             GenomicAlignments_1.14.1
[34] bitops_1.0-6             grid_3.4.2               xtable_1.8-2            
[37] gtable_0.2.0             DBI_0.7                  magrittr_1.5            
[40] scales_0.5.0             stringi_1.1.6            reshape2_1.4.2          
[43] doParallel_1.0.11        RColorBrewer_1.1-2       iterators_1.0.8         
[46] tools_3.4.2              bit64_0.9-7              PMCMR_4.1               
[49] AnnotationDbi_1.40.0     colorspace_1.3-2         memoise_1.1.0           

```

**Comment by rvalieris** [comment]
thanks for the detailed explanation @ulah,
another workaround is to call VA functions by the full name like: VariantAnnotation::readVcf, then you don't need to `library(VariantAnnotation)`.

I need to think of a more permanent fix for this so I will leave this issue open for now.

**Comment by rvalieris** [comment]
this should be fixed for good on the new version released with bioconductor 3.7.

let me know if the problem arises again.
